tests/std/tests/LWG3528_make_from_tuple_impl/test.compile.pass.cpp

```diff
@@ -0,0 +1,103 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//===----------------------------------------------------------------------===//
+
+#include <cstdint>
+#include <tuple>
+#include <type_traits>
+
+struct A {
+    int a;
+};
+struct B : public A {};
+
+struct C {
+    C(const B&) {}
+};
+
+enum class D {
+    ONE,
+    TWO,
+};
+
+template <class T, class Tuple>
+auto can_make_from_tuple(T&&, Tuple&& t) -> decltype(std::make_from_tuple<T>(t), uint8_t()) {
+    return 0;
+}
+template <class T, class Tuple>
+uint32_t can_make_from_tuple(...) {
+    return 0;
+}
+
+template <class T, class Tuple>
+inline constexpr bool has_make_from_tuple =
+    std::is_same_v<decltype(can_make_from_tuple<T, Tuple>(T{}, Tuple{})), uint8_t>;
+
+template <class T, class Tuple>
+auto can_make_from_tuple_impl(T&&, Tuple&& t)
+    -> decltype(std::_Make_from_tuple_impl<T>(std::forward<Tuple>(t),
+                    std::make_index_sequence<std::tuple_size_v<std::remove_reference_t<Tuple>>>{}),
+        uint8_t()) {
+    return 0;
+}
+template <class T, class Tuple>
+uint32_t can_make_from_tuple_impl(...) {
+    return 0;
+}
+
+template <class T, class Tuple>
+inline constexpr bool has_make_from_tuple_impl =
+    std::is_same_v<decltype(can_make_from_tuple_impl<T, Tuple>(T{}, Tuple{})), uint8_t>;
+
+// Test std::make_from_tuple constraints.

```

Comment:
Not sure whether this would be valuable. We can additionally test using `std::array<U, 1>` as tuple-like types.

Reply:
Agree, I've added test for `std::array<U, 1>`.